### PR TITLE
Update go tour documentation to reflect dataset changes

### PR DESCRIPTION
**Pull request by wardn**

### CODE REVIEW

File: doc/go-tour.md [hunk 1]
```diff
@@ -117,12 +116,11 @@ func newPerson(givenName string, male bool) types.Struct {
 }
 
 func main() {
-  db, ds, err := spec.GetDataset("http://localhost:8000::people")
+  sp, err := spec.ForDataset("http://localhost:8000::people")
   if err != nil {
     fmt.Fprintf(os.Stderr, "Could not create dataset: %s\n", err)
     return
   }
-  defer db.Close()
```
Comment (kalman) [inline doc/go-tour.md]: Still need `sp.Close()`

File: doc/go-tour.md [hunk 2]
```diff
@@ -164,14 +162,13 @@ import (
 )
 
 func main() {
-  db, ds, err := spec.GetDataset("http://localhost:8000::people")
+  sp, err := spec.ForDataset("http://localhost:8000::people")
   if err != nil {
     fmt.Fprintf(os.Stderr, "Could not create dataset: %s\n", err)
     return
   }
-  defer db.Close()
```
Comment (kalman) [inline doc/go-tour.md]: `sp.Close()`

File: doc/go-tour.md [hunk 3]
```diff
@@ -232,14 +229,13 @@ import (
...
```
Comment (kalman) [inline doc/go-tour.md]: Here too

File: doc/go-tour.md (hunk 1 above)
Comment (kalman) [inline doc/go-tour.md]: `defer sp.Close()` I mean.

**Comment by wardn** [comment]
Now deferring all spec closes.

**Pull request merged**